deno_dist/middleware/cache/index.ts

```diff
@@ -5,28 +5,60 @@ export const cache = (options: {
   cacheName: string
   wait?: boolean
   cacheControl?: string
+  vary?: string
 }): MiddlewareHandler => {
   if (options.wait === undefined) {
     options.wait = false
   }
 
-  const directives = options.cacheControl?.split(',').map((directive) => directive.toLowerCase())
+  const cacheControlDirectives = options.cacheControl
+    ?.split(',')
+    .map((directive) => directive.toLowerCase())
+  const varyDirectives = options.vary?.split(',').map((directive) => directive.trim())
+  // RFC 7231 Section 7.1.4 specifies that "*" is not allowed in Vary header.
+  // See: https://datatracker.ietf.org/doc/html/rfc7231#section-7.1.4
+  if (varyDirectives?.includes('*')) {
+    throw new Error(
+      'Middleware vary configuration cannot include "*", as it disallows effective caching.'
+    )
```

Comment:
> A proxy MUST NOT generate a Vary field with a "*" value.

cf: https://datatracker.ietf.org/doc/html/rfc7231#section-7.1.4

Seeing this in the RFC, we now throw when `Vary` is set to `*`.

Please check this.